tests/reftests/shared-fetch.test

```diff
@@ -0,0 +1,117 @@
+N0REP0
+### <shared.ml>
+print_endline "I am a shared code"
+### <non-shared.ml>
+print_endline "I am a non shared code"
+### tar czf shared.tgz shared.ml
+### tar czf non-shared.tgz non-shared.ml
+### <pkg:no-extra.3>
+opam-version: "2.0"
+build: [ "ocaml" "shared.ml" ]
+### <pkg:extra.3>
+opam-version: "2.0"
+build: [
+  [ "ocaml" "shared.ml" ]
+  [ "ocaml" "extra_uniq.ml" ]
+]
+#patches: ["shared.patch"]
+### <pkg:extra.3:extra_uniq.ml>
+print_endline "I am a non shared code"
+### <pkg:non-shared.4>
+opam-version: "2.0"
+build: [ "ocaml" "non-shared.ml" ]
+### <mkurl.sh>
+arch=$1
+shift
+for p in $@; do
+  file="REPO/packages/${p%.*}/$p/opam"
+  echo "url {" >> $file
+  echo "src: \"${BASEDIR}/$arch.tgz\"" >> $file
+  MD5=$(openssl md5 $arch.tgz | cut -d' ' -f2)
+  echo "checksum: \"md5=$MD5\"" >> $file
+  echo "}" >> $file
+done
+### sh mkurl.sh shared no-extra.3 extra.3
+### sh mkurl.sh non-shared non-shared.4
```

Comment:
It looks like this test needs to be duplicated for it to pass on (non-cygwin) Windows
```
diff --git a/tests/reftests/shared-fetch.test b/tests/reftests/shared-fetch.out
index faf44ff3..2bcf991a 100644
--- a/tests/reftests/shared-fetch.test
+++ b/tests/reftests/shared-fetch.out
@@ -38,22 +38,17 @@ done
 
 <><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
 [default] synchronised from file://${BASEDIR}/REPO
-Now run 'opam upgrade' to apply any package updates.
+[ERROR] At ${OPAMTMP}/default/packages/extra/extra.3/opam:8:9-8:9::
+        illegal escape sequence
+[ERROR] Strict mode: aborting
+[ERROR] Could not update repository "default": OpamStd.OpamSys.Exit(30)
+# Return code 40 #
 ### opam switch create shared --empty
 ### opam install no-extra extra non-shared | unordered
-The following actions will be performed:
-  - install extra      3
-  - install non-shared 4
-  - install no-extra   3
-===== 3 to install =====
-
-<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
--> retrieved extra.3+no-extra.3  (file://${BASEDIR}/shared.tgz)
--> retrieved non-shared.4  (file://${BASEDIR}/non-shared.tgz)
--> installed extra.3
--> installed no-extra.3
--> installed non-shared.4
-Done.
+[ERROR] No package named no-extra found.
+[ERROR] No package named extra found.
+[ERROR] No package named non-shared found.
+# Return code 5 #
```

Reply:
It seems to come from `${BASEDIR}`, cf  [this run](https://ci.appveyor.com/project/rjbou/opam/builds/41459574/job/emefj30g2gcif7eo#L2673)
```
### cat REPO/packages/extra/extra.3/opam
+opam-version: "2.0"
+build: [
+  [ "ocaml" "shared.ml" ]
+  [ "ocaml" "extra_uniq.ml" ]
+]
+#patches: ["shared.patch"]
+url {
+src: "${BASEDIR}/shared.tgz"
+checksum: "md5=a13945d953fc7ab20f4f0b80f00211cb"
+}